athena_udf/__init__.py

```diff
@@ -35,9 +42,25 @@ def handle_udf_request(cls, event):
         record_batch_list = record_batch.to_pylist()
 
         outputs = []
-        for record in record_batch_list:
-            output = cls.handle_athena_record(input_schema, output_schema, list(record.values()))
-            outputs.append(output)
+        with ThreadPoolExecutor() as executor:
```

Comment:
just a thought about this? what if the user don't want to use the ThreadPoolExecutor, for whatever reason.
IMHO we should let the user decide to use or not threads. We can consider to add another paramter called: `use_thread: boolean`, to let the user decide if to use or not threads.
We can use a default value to True, to force thread usage.

Reply:
I thought about that. So in case when threading is not in use, we don't need chunking as well. WDYT ?

Reply:
I don't think so. So the code to have 3 fork for these use cases:
* use_thread:false - (no chunk) - pretty much the original implementation
* use_thread: true, and chunk is not None
* use_thread: true and chunk is None

Reply:
definitely, this is what I meant. agreed then